### PR TITLE
Allows adding annotaiton to pilot service account through helm chart

### PR DESCRIPTION
**Pull request by troydai**
**Please provide a description of this PR:**

This change add a new option for users of the istiod helm chart to add annotation to the istiod's
service account.

If a user system relies on service account annotation to trigger automation such as SPIRE workload
registeration, this can be useful.

### CODE REVIEW

**Comment by zirain** [comment]
why this's a must? you can try https://istio.io/latest/docs/setup/additional-setup/customize-installation-helm/.

**Comment by troydai** [comment]
This does make it simpler without requiring yet another process/rendering in the process. The default helm chart already has service annotations and pod annotations configurable through the template itself, why not adding the set of annoation?

**Comment by troydai** [comment]
Some users may have build-pipelines managed by a separate organization or have a unique setup that makes applying post-render tricky (or simply harder to disentangle in some situations). Allowing this expansion is a low-cost way to improve the usability of the helm charts.

**Comment by bleggett** [comment]
I'm not against this - in the last combined WG @dgn proposed that we support basic overrides for all charts - I would consider setting annotations to be "basic overrides".

https://docs.google.com/document/d/1XY8OvC6Cc4iSlVLCDW3ghatV1cftX2kXU4y61Btqiyw/edit?resourcekey=0-cnSty1dQShh7mRWxJxJUNQ#heading=h.5egfgy2fhcii

(the document says "we should support these overrides for all top-level resources" and SvcAcct is a top-level resource, so this makes sense to me)

**Comment by bleggett** [comment]
@troydai CLA and release notes please

**Comment by troydai** [comment]
Signed CLA and add release note.

**Comment by troydai** [comment]
I'll spend some time in the next few days to fix up the tests.

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -56,6 +56,7 @@ defaults:
     nodeSelector: {}
     podAnnotations: {}
     serviceAnnotations: {}
+    serviceAccountAnnotations: {}
```
[the remark this answers is not in the document]
Comment (troydai) [reply manifests/charts/istio-control/istio-discovery/values.yaml:59]: You already can add annotation on gateway's service account
https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/serviceaccount.yaml#L19

What's the inconsistency of application of what should be added?

If we need to draw a line, how about at basic build blocks: pod, service, service account?

[the remark this answers is not in the document]
Comment (hanxiaop) [reply manifests/charts/istio-control/istio-discovery/values.yaml:59]: Thanks, this basic override makes sense to me now. I had thought the gateway was using the general annotations applied to all resources.

**Review by costinm — APPROVED** [review]
Seems reasonable, will be very useful for GKE workload identity.

**Comment by troydai** [comment]
> Seems reasonable, will be very useful for GKE workload identity.

Thanks. I need to rebase nevertheless.

**Review by jacob-delgado — APPROVED** [review]

**Comment by troydai** [comment]
I'm a little bit puzzled by the CI failures. Glance at the logs, I can figure how it related to the change. Will take a further look when I get a chance.

**Comment by zirain** [comment]
> I'm a little bit puzzled by the CI failures. Glance at the logs, I can figure how it related to the change. Will take a further look when I get a chance.

it's flakes, see https://github.com/istio/istio/issues/49793

**Pull request merged**